internal/goeland/fetch/source.go

```diff
@@ -11,10 +11,13 @@ import (
 )
 
 // FetchSource retrieves a source from either a feed, imgur or other sub-sources
-func FetchSource(config config.Provider, sourceName string) (*goeland.Source, error) {
+func FetchSource(config config.Provider, sourceName string, parents []string) (*goeland.Source, error) {
 	if !config.IsSet(fmt.Sprintf("sources.%s", sourceName)) {
 		return nil, fmt.Errorf("cannot find source: %s", sourceName)
 	}
+	if filters.StringInSlice(strings.ToLower(sourceName), parents) {
+		return nil, fmt.Errorf("source: %s has already be seen, preventing cycle", sourceName)

```

Comment:
"source: %s has already be**en** seen, preventing cycle"

But I think a better error message for the end user would be something like:
"there seems to be a cycle in your configuration. Make sure no "merge" source is referencing itself."

It gives out the information that there is a problem with the user's config and doesn't induce the user into thinking the `source %s` is the problem because it's not, it just happens to be the first source that was fetched twice.

Reply:
Agreed